File: 4.rb
#!/usr/bin/env ruby
#A palindromic number reads the same both ways.
#The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
#
#Find the largest palindrome made from the product of two 3-digit numbers.
answer = 0
(100..999).each do |a|
  (100..999).each do |b|
    num = a * b
    if num.to_s.eql?(num.to_s.reverse) && num > answer
      answer = num
    end
  end
end

puts answer

File: c/4.c
/* A palindromic number reads the same both ways. The largest palindrome
 * made from the product of two 2-digit numbers is 9009 = 91 × 99.
 * Find the largest palindrome made from the product of two 3-digit numbers.*/
#include <stdio.h>
#include <math.h>
/* determine whether or not the product of these numbers
 * is palnindromic */
int is_palindromic(int x, int y){
}
int main(void){
  int a = 999;
  int b = 999;
  for(a = 999;a => 100; --a){
    for(b = 999;b => 100; --b){
      if(is_palindromic(a,b) == 1)
        /* keep track of the highest palindromic numbers */
        break;
    }
  }
}
